src/vector/backends/awkward_constructors.py

```diff
@@ -256,7 +287,14 @@ def Array(*args: typing.Any, **kwargs: typing.Any) -> typing.Any:
     import vector
     import vector.backends.awkward  # noqa: 401
 
-    akarray = awkward.Array(*args, **kwargs)
+    layout = awkward.to_layout(args[0])
+    array_type = awkward.type(layout)
+    args = args[1:]
+
+    if not _is_type_safe(array_type):
+        raise TypeError("a coordinate must be of the type int or float")
+
+    akarray = awkward.Array(layout, *args, **kwargs)
```

Comment:
```suggestion
    akarray = awkward.Array(*args, **kwargs)
    array_type = akarray.type

    if not _is_type_safe(array_type):
        raise TypeError("a coordinate must be of the type int or float")
```

You didn't need to make the `layout`, but in doing so you would have lost some ways that `behavior` could pass through to the final output and you weren't handling the `args` and `kwargs` safely: a user could pass the `data` as a keyword; it's not a position-only argument.

The `awkward.type` function works on `akarray` as well as `layout`, but when we guarantee that it's an `ak.Array`, we can even use its `type` property, no special function needed.